### DISCUSSION TITLE
Morphing Digits

### DISCUSSION THREAD

**Discussion by rdulak** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#110] [Q&A] [answered]
Hello 😀.
You are great. I get a lot of fun from your libraries.
Could you help me customize the MorphDigit library to work with your library ? I got the correct work but only for the basic mode. Unfortunately for virtual mode I can't cope with the classes.
Thank you ☺️

https://github.com/hwiguna/HariFun_165_Morphing_Digits

**Comment by mrcodetastic** [comment]
Hi Rdulak,

Thanks for the kind words.

Not sure any help can be provided here as the morph clock is hard coded for
a 64x32px matrix only.

You'll need to spend time and experiment.



>

**Comment by bogd** [comment, marked as answer]
@rdulak - I have a working version of the clock on a 128x64 matrix, using this library (it did require some modifications to the code, but nothing major). 

You can find the full project (code, PCB, case) [here](https://github.com/bogd/esp32-morphing-clock). The instructions are a bit lacking at the moment (I didn't have the time to write full instructions for everything), but maybe this will help with your project?

The relevant files should be digit.h/digit.cpp, and rgb_display.h/rgb_display.cpp

**Reply by vortigont** [reply]
Nice clock indeed. :) Needs a WebUI to make all settings easier.

**Reply by mrcodetastic** [reply]
That's epic @bogd. Thanks for sharing!

**Reply by bogd** [reply]
:) Thank you for the kind words. It is still very much a work in progress, and I appreciate any suggestions that could help me improve it. 

Speaking of which: @vortigont , that is indeed a very good suggestion - thank you! Unfortunately, I have zero experience on that side (implementing a web UI), so I do not even know where to start. I will try to look at some documentation, but for now my main focus is implementing the planned features (weather forecast, alerts, etc). So it might be a while until I get around to the webUI thing (especially considering that until now the main focus was just getting the @#$@ thing to work - the first iteration of the PCBs was unusable :( ).

@mrfaptastic - thank you once again for making this possible! If you wish, you can add my PCB to the list of available PCBs for your library (now that I finally received and tested the new version, I can confidently say that it works correctly :) ).

**Reply by vortigont** [reply]
@bogd there is a [framework](https://github.com/vortigont/EmbUI) that could help in creating nice WebUI controls for esp's. And it also has some most common things integrated already, like WiFi manager, NTP, TimeZones, mdns etc, Not well documented, but any way... :) You can check one of my [projects](https://github.com/vortigont/espem) using this framework if you like.
Cheers!
